### PR TITLE
Add formic module to game

### PR DESCRIPTION
**Pull request by Steampunkery**
This module adds a ship, formic small to the game.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

**Comment by vampcat** [comment]
Okay, so I really like the ship, but I can't say the same about the PR.

I'm not sure what workflow you're using (eclipse? cmd? intellij? something else?) but adding all the files back to /src/main/java is definitely not what we want to do - efforts have gone into following the same project structure as Terasology, and the old structure (that your PR re-introduces) is not the way to go.
So, I recommend you get on irc/slack sometime, and figure out the problem with your workspace so we can continue working without having to add files here :)
(as an added bonus, we might be able to support eclipse/whatever properly too, so i definitely encourage it)

**Comment by Sigma-One** [comment]
I also like the ship a lot. It looks like it is made with that one modular ship pack from Open Game Art, so make sure you aren't breaking the license.

**Comment by Steampunkery** [comment]
@Sigma-One It is one of the ones from Open Game Art. The license is CC-BY-SA which (as I understand) means we can use it for anything without giving credit.

**Comment by vampcat** [comment]
Let's see what our resident bot has to say about you.
@GooeyHub: ok to test

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Cervator** [comment]
> It is one of the ones from Open Game Art. The license is CC-BY-SA 3.0 which (as I understand) means we can use it for anything without giving credit.

Not quite. BY means it *does* require attribution and is our default. The SA means "share-alike" which means it *must* be shared the same way, as in we couldn't take it closed source or anything (which you *can* do with BY or even Apache v2 for code)

Whenever possible it helps to use BY alone since the SA makes changing things harder if we should want to one day. But in modules it isn't too bad - we could always just walk away from it one day. Engine type stuff should always be Apache 2 + BY

File: .gitignore [hunk 1]
```diff
@@ -41,6 +41,7 @@ tmp/
 *.bak
 *.swp
 *~.nib
+desktop/src/main/resources
```
Comment (Cervator) [inline .gitignore]: Unsure why this is (still) in here :-)

File: engine/src/main/java/org/destinationsol/ModuleManager.java [hunk 2]
```diff
@@ -94,6 +94,7 @@ public ModuleManager() {
         requiredModules.add(registry.getLatestModuleVersion(new Name("core")));
         requiredModules.add(registry.getLatestModuleVersion(new Name("federal")));
         requiredModules.add(registry.getLatestModuleVersion(new Name("organic")));
+	requiredModules.add(registry.getLatestModuleVersion(new Name("formic")));
```
Comment (Cervator) [inline engine/src/main/java/org/destinationsol/ModuleManager.java]: Whitespace is off here. Make sure you're using IntelliJ/Atom with comparable settings (some sort of smart adaption where it'll match existing whitespace or hard set to 4-space indentation)

File: modules/formic/assets/ships/formicSmall/formicSmall.json [hunk 3]
```diff
@@ -0,0 +1,1164 @@
+{
```
Comment (Cervator) [inline modules/formic/assets/ships/formicSmall/formicSmall.json]: Whitespace seems off in this file. Maybe bulk-format it? 4 spaces per indentation layer to match the other JSON files including in the same PR :-)

File: modules/formic/assets/ships/formicSmall/formicSmall.json [hunk 4]
```diff
@@ -0,0 +1,1164 @@
...
+          },
+          {
+            "x": 0.2730769217014313,
+            "y": 0.6346153616905212
+          }
+        ]
+      }
+    ]
+  }
+}
```
Comment (Cervator) [inline modules/formic/assets/ships/formicSmall/formicSmall.json]: Missing newline at the end of the file (bulk reformat might handle this but keep a tiny eye out for this and other good practices)

File: .gitignore (hunk 1 above)
Comment (Steampunkery) [reply .gitignore]: That’s there because there something wrong with the path for resources, and I didn’t want it to get pushed in the PR.

Comment (Cervator) [reply .gitignore]: Oh you can just choose not to commit those files if your workspace has gone weird. The path itself is perfectly valid - there just aren't any actual resources for the desktop sub-project (yet). So we shouldn't gitignore it.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by vampcat** [comment]
Okay, so I'm not a hundred percent sure about the Medium1 file. Is it required?

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by vampcat** [comment]
I'm not sure how much shot-gun this is, and how much machine-gun.. but, I like it :D

**Comment by Steampunkery** [comment]
Is more like an auto shotgun

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Pull request closed without merging**